### DISCUSSION TITLE
Is there anything available for hierarchical relations?

### DISCUSSION THREAD

**Discussion by ajmal-yazdani** [Gremlinq/ExRam.Gremlinq#783] [Q&A] [answered]
In gremlin I can travers to parent hierarchies through below "repeat - out" query. Is there anything available with Gremlinq? 

g.V().has('name', 'D').repeat(out('parent')).emit().values('name')

Thanks.

**Comment by danielcweber** [comment, marked as answer]
There's the Loop-method that let's you build all kinds of repeat-until-emit constructs. See a unit test [here](https://github.com/ExRam/ExRam.Gremlinq/blob/9d2eb23efed04db311eed37ef626decc11447499/test/ExRam.Gremlinq.Core.Tests/QueryExecutionTest.cs#L3812).
